js_modules/dagit/packages/ui/src/components/ProductTour.tsx

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import {createPortal} from 'react-dom';
+import styled, {CSSProperties} from 'styled-components/macro';
+
+import {Box} from './Box';
+import {Button} from './Button';
+import {Colors} from './Colors';
+import {Subheading} from './Text';
+
+export enum Position {
+  TOP_LEFT,
+  TOP_CENTER,
+  TOP_RIGHT,
+  BOTTOM_LEFT,
+  BOTTOM_CENTER,
+  BOTTOM_RIGHT,
+}
+
+type ObjectType =
+  | {img: string; video?: undefined; object?: undefined}
+  | {video: string; img?: undefined; object?: undefined}
+  | {object: React.ReactNode; video?: undefined; img?: undefined}
+  | {img?: undefined; video?: undefined; object?: undefined};
+
+type Props = {
+  title: React.ReactNode;
+  description?: React.ReactNode;
+  actions?: {
+    next?: () => void;
+    dismiss?: () => void;
+  };
+  position: Position;
+  width?: CSSProperties['width'];
+} & ObjectType;
```

Comment:
Do the `ObjectType` fields need to be intersected onto `Props`? Or could the image/video/object be passed in as a single prop of `ObjectType`?

Reply:
The name of the field is how I determine which type it is image/video/object.

Otherwise you'd need to pass the type in some other way, I think using the name of the prop is the cleanest from an API perspective.